pkg/trace/call.go

```diff
@@ -107,6 +106,13 @@ func SetCallError(ctx context.Context, err error) error {
 // rethrows them.  Any panics are converted to errors and cause error
 // logging to happen (as do any SetCallStatus calls)
 func EndCall(ctx context.Context) {
+	callInfo := callTracker.Info(ctx)
+	if callInfo == nil {
+		// There was no call associated with this context, so we can't
+		// do anything.
+		return
+	}
+
 	defer End(ctx)
```

Comment:
I think you should still have this deferred even with a nil `callInfo`, to try to maintain compatibility as much as possible. This function doesn't rely on the info, only the span, so it wouldn't have panicked before.

Reply:
If `callInfo` is `nil`, this does panic actually. `End()` ends up using it later in the call stack. I don't know of an easy way to avoid that yet, but just noting that it is somewhat dangerous right now.

Reply:
I was wrong, `callTracker.EndCall` does, not `End` itself.